contracts/core/SequencerSoloClient-ancestorProof.sol

```diff
@@ -0,0 +1,153 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+pragma solidity 0.8.15;
+
+import {Ics23Proof} from "../interfaces/IProofVerifier.sol";
+import {ISignatureVerifier} from "../interfaces/ISignatureVerifier.sol";
+import {ILightClient, LightClientType} from "../interfaces/ILightClient.sol";
+import {L1Block} from "optimism/L2/L1Block.sol";
+import {console2} from "forge-std/console2.sol";
+
+/**
+ * @title SubfinalityLightClient
+ * @author Polymer Labs
+ * @dev This specific light client implementation uses the same client that is used in the op-stack
+ */
+contract SequencerSoloClientAncestorProof is ILightClient {
+    error InvalidL1Origin();
+
+    LightClientType public constant LIGHT_CLIENT_TYPE = LightClientType.SequencerLightClient; // Stored as a constant
+        // for cheap on-chain use
+    uint8 private _LightClientType = uint8(LIGHT_CLIENT_TYPE); // Also redundantly stored as a private mutable type in
+        // case it needs to be accessed in any proofs
+
+    // consensusStates maps from the height to the appHash.
+    mapping(uint256 => uint256) public consensusStates;
+    mapping(bytes32 => bool) public checkpoints;
+
+    ISignatureVerifier public immutable verifier;
+    L1Block public l1BlockProvider;
+
+    /// Optimization: for the l1 block hash, we can use a ring buffer. reading from this can be done off chain,, so
+    /// relayer provides the l1 index
+    // SO we need a getter for that option.
+    // And then we just look at the block index at this poitn. and then we can remove this/overwrite blockhashes for it.
+    // to avoi
+    // Always autoincrement the ring buffer, so that you can always auto-increment the prev hash.
+    // Note; we need to think about the case where we have a griefer for the ring buffer.
+    // Doing this in a hash map is probably nicer here.
+    //
+    error CannotUpdatePendingOptimisticConsensusState();
+    error AppHashHasNotPassedFraudProofWindow();
+    error NonMembershipProofsNotYetImplemented();
+    error NoConsensusStateAtHeight(uint256 height);
+
+    constructor(ISignatureVerifier verifier_, L1Block _l1BlockProvider) {
+        verifier = verifier_;
+        l1BlockProvider = _l1BlockProvider;
+    }
+
+    /**
+     * @inheritdoc ILightClient
+     * @param proof An array of bytes that contain the l1blockhash and the sequencer's signature. The first 32 bytes of
+     * this argument should be
+     * the l1BlockHash, and the remaining bytes should be the sequencer signature which attests to the peptide AppHash
+     * for that l1BlockHash
+     */
+    function updateClient(bytes calldata proof, uint256 peptideHeight, uint256 peptideAppHash) external override {
+
+        // Here proof will contain the signed header like in the previous one. Then we will need to pass in the  
+        // the length of block heights from 1 to others. 
+
+        console2.log("length");
+        console2.log(proof.length);
```

Comment:
remove debugging logs?

Reply:
I removed this contract from this PR altogether